src/views/List.jsx

```diff
@@ -1,15 +1,61 @@
 import { ListItem } from '../components';
+import { useEffect, useState } from 'react';
 
 export function List({ data }) {
+	const [searchQuery, setSearchQuery] = useState('');
+	const [searchResults, setSearchResults] = useState([]);
+	// user type in search item - query
+	// as letters come in, filters items rendered on page
+	// set query to searchQuery
+	// render onto page search Results -> setting whatever is filtered into searchResults
+
+	function filterResults(query) {
+		return data.filter((item) =>
+			item.name.toLowerCase().includes(query.toLowerCase()),
+		);
+	}
+
+	useEffect(() => {
+		setSearchResults(filterResults(searchQuery));
+
+		// ignoring dependency array warning for now
+		// adding filterResults causes infinite re-render
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [searchQuery]);
+
+	function handleClick() {
+		setSearchResults([]);
+	}
+
 	return (
 		<>
 			<p>
 				Hello from the <code>/list</code> page!
 			</p>
+			<form>
+				<label htmlFor="search-items">
+					Search Items:
+					<input
+						name="search-items"
+						id="search-items"
+						type="text"
+						placeholder="Search items"
+						value={searchQuery}
+						onChange={(e) => setSearchQuery(e.target.value)}
+					/>
+					{searchQuery ? (
+						<button onClick={handleClick}>Clear search</button>
```

Comment:
This button is defaulting to type="submit" which is the default when a button is inside a form I think we should specify type="button" here. What's happening is it's submitting the search query input value as a URL query and we don't use URL querying and then also doing the default submit event behavior of causing page reload and I don't think we want any of that UX.

In this screenshot you can see at the end of the URL that search-items=trea which is currently happening when I click this button.
![handleClick](https://user-images.githubusercontent.com/89806097/180387901-30c9f39f-36b6-4297-b356-17e1441375c7.PNG)


https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button

Reply:
Learned something new today- thank you!